### PR TITLE
Adding default to save relative paths

### LINKED ISSUE #35: Remove "github/workspace" from filename in csv?
Not sure if this belongs to the library or action. Would it be possible (and make sense) to have the filename relative to the folder in which the URL checks are run?

### PR DESCRIPTION
**Pull request by vsoch**
This will add a default to the save_results function to save relative paths (relative_paths=True). Since this is considered advanced functionality, it isn't exposed via the client, but instead is part of the class for the advanced user to change, if needed. This will close #35, and be distributed in a future release.

Signed-off-by: vsoch <vsochat@stanford.edu>

### CODE REVIEW

**Comment by vsoch** [comment]
Going to push another test.

**Comment by vsoch** [comment]
Going back to sleep for a bit... will check this and adjust any bugs/fails when I wake up (bad to program in the middle of the night...) :)

File: tests/test_core_check.py [hunk 1]
```diff
@@ -111,6 +113,20 @@ def test_check_run_save(tmp_path, retry_count):
     # Ensure content looks okay
     for line in lines[1:]:
         url, result, filename = line.split(",")
+
+        root = filename.split('/')[0]
         assert url.startswith("http")
         assert result in ["passed", "failed"]
         assert re.search("(.py|.md)$", filename)
```
Comment (SuperKogito) [inline tests/test_core_check.py:120]: I noticed that you are using a lot of `assert ...``which will cause some warnings from Codacy, mentioning that it is better to use: ```if not ...: raise AssertionError `. We can keep it the way it is for now, but -I think- we will probably need to change this in the future.

File: tests/test_files/sample_test_file.md [hunk 2]
```diff
@@ -5,27 +5,15 @@ The following is a list of test urls to extract.
...
-- [test url 20](https://github.com/SuperKogito/Voice-based-gender-recognition)
-- [test url 21](https://github.com/SuperKogito/Voice-based-gender-recognition/issues)
-- [test url 22](https://github.com/SuperKogito/Voice-based-gender-recognition/issues/1)
+- [test url 10](https://github.com/SuperKogito/Voice-based-gender-recognition)
+- [test url 11](https://github.com/SuperKogito/Voice-based-gender-recognition/issues)
 - [test url 23](https://github.com/SuperKogito/Voice-based-gender-recognition/issues/2)
-- [test url 24](https://github.com/SuperKogito/Voice-based-gender-recognition/issues)
 
 - [broken_test url 1](https://none.html)
 - [broken_test url 2](https://github.com/SuperKogito/URLs-checker/README.md)
```
Comment (SuperKogito) [inline tests/test_files/sample_test_file.md:19]: not that I am against it but Is there a particular reason for deleting a bunch of links?

File: tests/test_core_check.py (hunk 1 above)
Comment (vsoch) [reply tests/test_core_check.py:120]: Ok! Sorry about that. We can definitely change it in the future, I’ve mostly been avoiding warnings from pytest and these seemed ok.

File: tests/test_files/sample_test_file.md (hunk 2 above)
Comment (vsoch) [reply tests/test_files/sample_test_file.md:19]: For this particular set of tests, having the extra urls (with retry) didn’t benefit the test but just made it longer, so I removed some that were redundant. It definitely could still be improved upon!

**Comment by vsoch** [comment]
Awesome! Let’s merge and I’ll try to test your regex PR today and we can release a version after that on pypi with both added features.

**Comment by vsoch** [comment]
Whenever looks good, go ahead and approve and merge.

**Comment by vsoch** [comment]
Two quick questions for you before we merge:

 - should the variable be exposed to the client (e.g., --save_relative_paths and thus in the action)
 - should the default be True (use relative paths?)

**Comment by SuperKogito** [comment]
I just approved the PR. 
Well as I mentioned before, I prefer limiting the variables exposed to the client/user to a sufficient number.  So the current version where we don't expose the feature to the client and just use it internally via the default value seems adequate to me. Unless you disagree?  The only issue in this case might arise in the action but I think even in there the relative paths are a better option.

**Comment by vsoch** [comment]
I'm in agreement! This feels like a "choose the best default and don't expose until someone actually asks for it." Let's merge this guy! I'll be able to review your PR soon, have a group meeting and then some free time.

**Pull request merged**